source/maze/MazeIO.cpp

```diff
@@ -42,6 +42,22 @@ void MazeIO::setMousePosition(PositionController posCon)
 	mousePosition = posCon;
 }
 
+void MazeIO::setDestinations(std::vector<Position> des)
+{
+	destinations = des;
+}
```

Comment:
Since the position vector bigger than a few bytes passing object by value is slow. Please pass `des` by reference. Also add `const` qualifier.